src/atomate2/abinit/jobs/base.py

```diff
@@ -0,0 +1,815 @@
+        f"Cannot convert event_string of type {type(event_string)}. Should be a string."
+    )
+
+
+@dataclass
+class BaseAbinitMaker(Maker):
+    """
+    Base ABINIT job maker.
+
+    Parameters
+    ----------
+    name : str
+        The job name.
+    pseudos : list of str, PseudoTable
+        The pseudopotentials to use.
+    """
+
+    calc_type: str
+    input_set_generator: AbinitInputSetGenerator
+    name: str = "base abinit job"
+    pseudos: Union[
+        List[str], PseudoTable
+    ] = pseudo_dojo.OfficialDojoTable.from_djson_file(
+        os.path.join(
+            pseudo_dojo.dojotable_absdir("ONCVPSP-PBE-PDv0.4"), "standard.djson"
+        )
+    )
+    walltime: Optional[int] = None
+    CRITICAL_EVENTS: Sequence[str] = ()
+    dependencies: Optional[dict] = None
+    extra_abivars: dict = field(default_factory=dict)
+
+    # class variables
+    structure_fixed: ClassVar[bool] = True
+
+    def __post_init__(self):
+        """Process post-init configuration."""
+        self.critical_events = [
+            as_event_class(ce_name) for ce_name in self.CRITICAL_EVENTS
+        ]
+        if self.input_set_generator is None:
+            raise InitializationError("Input set generator is not set.")
+
+    @job
+    def make(
+        self,
+        structure: Optional[Structure] = None,
+        prev_outputs: Optional[Any] = None,
+        restart_from: Optional[Union[str, jobflow.Job]] = None,
+        # abinit_input_set: Optional[AbinitInputSet] = None,
+        # previous_abinit_input_set: Optional[AbinitInputSet] = None,
+        # restart_info=None,
+        history: Optional[JobHistory] = None,
+    ) -> Union[jobflow.Flow, jobflow.Job]:
+        """
+        Return an ABINIT jobflow.Job.
+
+        Parameters
+        ----------
+        structure : Structure
+            A pymatgen structure object.
+        prev_outputs : TBD
+            TODO: make sure the following is ok
+            An absolute path as a string (previous directory), a list of absolute paths as strings
+            (list of previous directories), a string (previous job uuid), a list of strings (list of previous job
+            uuids), an OutputReference or a list of OutputReference objects,
+            a dict with keys being a type of previous output and values being one of the previous (absolute
+            path or list of absolute paths, uuid or list of uuids, ...), ...
+        restart_from : TBD
+            An absolute path as a string (previous directory), a string (previous job uuid), an OutputReference,
+            TODO: what else ? is an AbinitInputSet ok ?
+            TODO: I would say no, because we not only need the input but also the location where it previously ran.
+        """
+        if structure is None and prev_outputs is None and restart_from is None:
+            raise RuntimeError(
+                "At least one of structure, prev_outputs or restart_from should be defined."
+            )
+
+        self.setup_job(
```

Comment:
I think it might be possible to rewrite the rest of this class for clarity by splitting the class methods into standalone functions with arguments. Fundamentally, there are 4 basic stages to the calculation:

1. Write input sets.
2. Run abinit
3. Read outputs.
4. Dynamic logic (e.g., resubmit?) 

All of the logic for writing input sets (creating files, folders etc) should be handled by the abinit input set. As much of possible, the logic for reading outputs should be in the AbinitTaskDocument. In this case there is logic for timing, logging, etc. I think that makes sense to go in the functions that actually run abinit.

I think the actual Maker should be quite simple. The `make` function should call a number of functions and then return the response. The Maker should have class variables for specifying overrides to the functions that get called. That way you can customise each job just by changing the class init variables.

You can see an example in the base VASP maker. There are several kwargs that include the function name they will be passed too: https://github.com/materialsproject/atomate2/blob/614ace0ee657759091b6e39296614b07b42140c8/src/atomate2/vasp/jobs/base.py#L110-L115

And then the make method is quite readable and easy to follow: https://github.com/materialsproject/atomate2/blob/614ace0ee657759091b6e39296614b07b42140c8/src/atomate2/vasp/jobs/base.py#L137-L163

I definitely appreciate that things will have to be a bit different for abinit, but I think we should follow the same pattern of having the `Makers` be easy to follow, with the complicated logic put into different functions organised into modules. For example, in the vasp module, there are a series of submodules:
- `run` containing all the logic for running vasp.
- `files` for containing IO logic.

Happy to chat about this further.

Reply:
Refactored